content/en/docs/Linters/links.md

```diff
@@ -5,74 +5,108 @@ weight: 10
+### Links to images
+*   Including an existing image - ![imageAPI](/Images/api.png).
 
-ISSUE(no-undefined-references thinks this is a problem) The HUGO's way of doing relative [links]({{< relref "../Guides/Best_Practices.md" >}}  "This is the link title")
+### Links to files
+*   Including a link to a json file that exists - [download apisecret.json](/samples/central/apisecret.json).
+*   Including a link to a js file that exists - [download config.js](/samples/snippets/config.js).
+*   Including a link to a yaml file that exists - [download apiservice.yaml](/samples/central/apiservice.yaml).
 
-Including an existing image - ![imageAPI](/Images/api.png).
+### Links to external locations
+*   An external location [link](https://google.co.uk/) that is accessible.
+*   An external file [link](https://docs.axway.com/bundle/API_Builder_4x_allOS_en/page/environmentalization.html) - verifying url replacement works as expected.
 
-Including a link to a file that exists - [download](/samples/central/apisecret.json).
+## "Working" links with lint errors
+This section contains working links that are not going to be used in our docs. We discourage the usage of such links and, fortunatelly, they are reported by the linter and can be changed to the format we prefer - pure markdown non relative links.
 
-An external [link](https://google.co.uk/) that is accessible.
+*   **no-undefined-references** for HUGO's way of doing [links]({{< ref "/docs/guides/debug_a_flow#debugging-flows" >}} "This is the link title").
+*   **no-undefined-references** for HUGO's way of doing relative [links]({{< relref "../Guides/Best_Practices.md" >}}  "This is the link title").
+*   **validate-internal-links** for a [link](../Guides/Best_Practices.md) using relative paths. WORKS in pure Markdown, does not work in HUGO. We do not want to use this style so we want linter warning for it.
+*   **validate-internal-links** for an [anchor](../Guides/Best_Practices.md#codebase) using relative paths. WORKS in pure Markdown, does not work in HUGO. We do not want to use this style so we want linter warning for it.
 
-An external [link](https://docs.axway.com/bundle/API_Builder_4x_allOS_en/page/environmentalization.html) - verifying url replacement works as expected.
+## Working reference style links
+This section contains working [reference style links][].
 
-A valid reference [google][]
+*   A valid [reference style link with definition at a bottom][].
+*   A valid reference [reference style link with inline definition][] - even though the reference is valid, the definition is expected to be at the bottom of the file, so we get **final-definition** linter warning for line 42.
 
-{{/* Even though the reference above is valid, the definition is expected to be at the bottom of the file, so we should receive a warning about this */}}
+[reference style link with inline definition]: https://google.co.uk/
```

Comment:
**Question**: why I am getting the warning at this line as below?
![image](https://user-images.githubusercontent.com/8027937/135615720-d65d7a59-3291-4987-8538-63c3feea7870.png)